common/djangoapps/util/tests/test_course_key_utils.py

```diff
@@ -5,28 +5,43 @@
 from util.course_key_utils import from_string_or_404
 from opaque_keys.edx.keys import CourseKey
 from django.http import Http404
+import ddt
+import unittest
 
 
-def test_from_string_or_404():
-
-    #testing with split style course keys
-    assert_raises(
-        Http404,
-        from_string_or_404,
-        "/some.invalid.key/course-v1:TTT+CS01+2015_T0"
-    )
-    assert_equals(
-        CourseKey.from_string("course-v1:TTT+CS01+2015_T0"),
-        from_string_or_404("course-v1:TTT+CS01+2015_T0")
+@ddt.ddt
+class TestFromStringOr404(unittest.TestCase):
+    """
+    Base Test class for from_string_or_404 utility tests
+    """
+    @ddt.data(
+        ("/some.invalid.key/course-v1:TTT+CS01+2015_T0", "course-v1:TTT+CS01+2015_T0"),  # split style course keys
+        ("/some.invalid.key/TTT/CS01/2015_T0", "TTT/CS01/2015_T0"),  # mongo style course keys
     )
+    def test_from_string_or_404(self, (invalid_course_key, valid_course_key)):
+        """
+        Tests from_string_or_404 for valid and invalid split style course keys and mongo style course keys.
+        """
+        assert_raises(
+            Http404,
```

Comment:
@Ayub-Khan you should be using `self.assertRaises` rather than nose `assert_raises` since you're using unit tests here.